src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -600,11 +600,9 @@ private List<GeneratorTemplatesList> getModElementListTemplates(ModElement eleme
 		List<?> templateLists = (List<?>) map.get("list_templates");
 		if (templateLists != null) {
 			int templateID = 0;
-			int listID = 1;
 			for (Object list : templateLists) {
 				Map<GeneratorTemplate, List<Boolean>> files = new LinkedHashMap<>();
-				String groupName = (String) Objects.requireNonNullElse(((Map<?, ?>) list).get("name"),
-						"Group " + listID);
+				String groupName = (String) ((Map<?, ?>) list).get("name");

```

Comment:
If you want this to be mandatory, it should still use a method in `Objects` but without the default value.